### PR TITLE
Issue #41 add outline shorthand

### LINKED ISSUE #41: Минимизация outline
<code>outline: none</code> → <code>outline: 0</code>

### PR DESCRIPTION
**Pull request by zoobestik**
resolves #41, I'm not sure that it is safe and still actual. But...

I maked these...

### CODE REVIEW

**Comment by lahmatiy** [comment]
Hi, @zoobestik 
This has little value for real CSS since `outline` doesn't used much. But we can add it.
Suggested implementation is incorrect. It breaks correct declarations like `outline: 1px none red`. `none` keyword can be replaced for `0` only when it's a single term in declaration's value.
I think optimisation should be applied for `border` property as well.

**Comment by zoobestik** [comment]
Hi, @lahmatiy,
I'm not sure I'm get clearly your message. These declarations shouldn't be broken.
* `outline: 1px none red` – for this declaration **node.children.head** property will be with `Dimension` type. And will be skipped by [if (value.type === 'Identifier')](https://github.com/css/csso/pull/332/files#diff-78a27c72637194de6ca6887ae71091e1R4)
* Other two `none red -> 0 red` and `none -> 0` – both look like correct options.

But all your corrections implemented. What do you think about it now?

File: test/fixture/compress/disjoin/1.min.css [hunk 1]
```diff
@@ -1,4 +1,4 @@
-a{color:red;width:200px}foo{border:none}b,foo{color:#00f;width:200px;height:50px}@media (min-width:400px){a{color:red;width:200px}foo{border:none}b,foo{color:#00f;width:200px;height:50px}}
+a{color:red;width:200px}foo{border:0}b,foo{color:#00f;width:200px;height:50px}@media (min-width:400px){a{color:red;width:200px}foo{border:0}b,foo{color:#00f;width:200px;height:50px}}
 /*!
   better solution
   a,b,foo{color:red;width:200px}foo{border:none}b,foo{color:#00f;height:50px}@media (min-width:400px){a,b,foo{color:red;width:200px}foo{border:none}b,foo{color:#00f;height:50px}}
```
Comment (lahmatiy) [inline test/fixture/compress/disjoin/1.min.css]: Please, change `none` -> `0` in comment too

File: test/fixture/compress/none-to-zero.css [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.o { outline: 1px none red }
+.o-none { outline: none }
+.o-none-red { outline: none red }
```
Comment (lahmatiy) [inline test/fixture/compress/none-to-zero.css]: Will be nice to add a test when `none` at the ending, e.g. `outline: red none`

File: lib/replace/property/border.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module.exports = function compressBorder(node) {
+    var value = node.children.head.data;
```
Comment (lahmatiy) [inline lib/replace/property/border.js]: Minor issue: it's better to store `node.children.head` as `child`

**Comment by lahmatiy** [comment]
> outline: 1px none red – for this declaration node.children.head property will be with Dimension type. And will be skipped by ...

Yep, but `outline: none 1px red` is not.

> Other two none red -> 0 red and none -> 0 – both look like correct options.

Agree.
I'd looked once again and realised that for `none red` we can remove `none` at all since it's a default. In general, if `none` is single term it can be replaced for `0`, otherwise it can be removed:

```js
value.children.each(function(node, item, list) {
    if (node.type === 'Identifier' && node.name.toLowerCase() === 'none') {
        if (list.head === list.tail) { // single term
            item.data = {
                type: 'Number',
                loc: node.loc,
                value: '0'
            };
        } else {
            list.remove(item);
        }
    } else if (node.type === 'WhiteSpace' && (item.prev === null || item.prev.type === 'WhiteSpace')) {
        list.remove(item)
    }
});
```

But it looks quite complicated. So you can fix my last comments and I'll merge PR.

**Comment by zoobestik** [comment]
🆙 Yep, it's looks really better.

File: test/fixture/compress/none-to-zero.css (hunk 2 above)
Comment (zoobestik) [inline test/fixture/compress/none-to-zero.css]: Done. `.o-red-none`one line below.

**Comment by lahmatiy** [comment]
Thank you for fixes.
I see issue with:
```css
.o-red-none { outline: red none }
```
after optimisation there is a redundant trailing whitespace
```css
.o-red-none{outline:red }
```

**Comment by lahmatiy** [comment]
My suggestion:

```js
function removeItemAndRedundantWhiteSpace(list, item) {
    var prev = item.prev;
    var next = item.next;

    if (next !== null) {
        if (next.type === 'WhiteSpace' && (prev === null || prev.type === 'WhiteSpace')) {
            list.remove(next);
        }
    } else if (prev !== null && prev.type === 'WhiteSpace') {
        list.remove(prev);
    }

    list.remove(item);
}

value.children.each(function(node, item, list) {
    if (node.type === 'Identifier' && node.name.toLowerCase() === 'none') {
        if (list.head === list.tail) {
            // replace `none` for zero when `none` is a single term
            item.data = {
                type: 'Number',
                loc: node.loc,
                value: '0'
            };
        } else {
            removeItemAndRedundantWhiteSpace(list, item);
        }
    }
});
```
Later I'll move `removeItemAndRedundantWhiteSpace` to helpers module.

**Comment by zoobestik** [comment]
🆙 Thanks you for your patience, this is my bad. I copied your snipped into my branch.

**Comment by lahmatiy** [comment]
Thank you!
